src/BindPattern.jl

```diff
@@ -131,12 +131,12 @@ function bind_pattern!(
         end
 
         # bind type at macro expansion time
-        (pattern0, assigned) = bind_pattern!(location, :(::($T)), input, state, assigned)
+        pattern0, assigned = bind_pattern!(location, :(::($T)), input, state, assigned)
         bound_type = (pattern0::BoundTypeTestPattern).type
         patterns = BoundPattern[pattern0]
 
         # TODO: Instead of calling fieldnames, we should call a method that can be
-        # overridden, e.g. by the implementation of `@auto_hash_equals_cached`, so
+        # overridden (propertynames?), e.g. by the implementation of `@auto_hash_equals_cached`, so
```

Comment:
In Julia, there's is `getproperty` and `propertynames` that are meant to be overloadable forms of dot-access. I'm not sure if that is what you're looking for here exactly, but thought I'd mention it.

Reply:
It would not help, because we want to generate the code based on the types, not the instances, so that we can generate static code once and for all.

Anyway, that comment will be obsolete once https://github.com/gafter/Rematch2.jl/pull/28 is merged.